### PR TITLE
20220211 stable version

### PR DESCRIPTION
**Pull request by ogarocious**
This is a stable version of the updates I created with the show pages of artists, events, dance styles, as well as adding filterrific, the youtube embeds, and some CSS styling to the custom.scss file.

### CODE REVIEW

File: app/controllers/artists_controller.rb [hunk 1]
```diff
@@ -44,7 +62,7 @@ def destroy
   private
 
   def artist_params
-    params.require(:artist).permit(:name, :description, :image_url, :instructor, :dj)
+    params.require(:artist).permit(:name, :description, :image_url, :instructor, :dj, :first_video_id, :second_video_id, :third_video_id, dance_style_ids: [], event_ids: [])
```
Comment (melchua) [inline app/controllers/artists_controller.rb:65]: I think you'll probably want to convert the videos param into one videos array. Probably best to make this fix so we don't have to change the database structure later. @ogarocious

File: app/views/artists/show.html.erb [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+
 <div class="container">
   <h2>Showing artist details</h1>
 
-  <div class="card" style="width: 18rem;">
-  <img src=<%= @artist.image_url %> class="card-img-top" alt="artist picture">
-  <div class="card-body">
-    <h5 class="card-title"><%= @artist.name %></h5>
-    <p class="card-text"><%= @artist.description %></p>
+  <div class="-">
```
Comment (melchua) [inline app/views/artists/show.html.erb]: what's this? that's a very odd name for a class name.

File: app/views/artists/show.html.erb [hunk 3]
```diff
@@ -1,18 +1,66 @@
...
+      </button>'.html_safe, edit_artist_path(@artist) %>
+
+      <%= link_to '<button type="button" class="btn btn-outline-danger mt-1">
+      Delete artist
+      </button>'.html_safe, artist_path(@artist), method: :delete, data: { confirm: "Are you sure?"} %>
+    <% end %>
+  </div>
+</div>
+</div>
+</div>
```
Comment (melchua) [inline app/views/artists/show.html.erb]: There's something odd here. I think the indentation is off, and maybe we have an extra `</div>`

File: app/views/dance_styles/show.html.erb [hunk 4]
```diff
@@ -1 +1,22 @@
...
+<h1 class="text-center mt-4">Dance Style: <%= @dance_style.name %></h1>
+
+<div class="event-list-wrapper";>
+<h3>Events</h3>
+  <% if @dance_style.events.any? %>
+<div class="row row-cols-4">
+      <%= render @dance_style.events %>
+  <% end %>
+</div>
+</div>
```
Comment (melchua) [inline app/views/dance_styles/show.html.erb]: Let's fix this indentation

File: app/views/dance_styles/show.html.erb [hunk 5]
```diff
@@ -1 +1,22 @@
...
+
+<div class="artist-list-wrapper";>
+<h3>Artists</h3>
+  <% if @dance_style.artists.any? %>
+<div class="row row-cols-4">
+      <%= render @dance_style.artists %>
+  <% end %>
+  </div>
+  </div>
```
Comment (melchua) [inline app/views/dance_styles/show.html.erb]: fix indentation

File: db/migrate/20220103143414_add_videoids_to_artists.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class AddVideoidsToArtists < ActiveRecord::Migration[6.1]
```
Comment (melchua) [inline db/migrate/20220103143414_add_videoids_to_artists.rb]: nitpick: `ids`, `I` should be capitalized

File: db/migrate/20220103143414_add_videoids_to_artists.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    def change
+      add_column :artists, :first_video_id, :string
+      add_column :artists, :second_video_id, :string
+      add_column :artists, :third_video_id, :string
```
Comment (melchua) [inline db/migrate/20220103143414_add_videoids_to_artists.rb]: We should probably aim to use an array here rather than separate video columns

**Review by melchua — CHANGES_REQUESTED** [review]
Hey the majority looks good! I did a quick code review and left some comments for changes to this branch. 

I haven't tested your branch yet, but will do so tmr!

File: app/views/artists/show.html.erb (hunk 2 above)
Comment (ogarocious) [reply app/views/artists/show.html.erb]: I think I just wanted to remove the class, I will double-check it

File: db/migrate/20220103143414_add_videoids_to_artists.rb (hunk 6 above)
Comment (ogarocious) [reply db/migrate/20220103143414_add_videoids_to_artists.rb]: Fixed!

File: db/migrate/20220103143414_add_videoids_to_artists.rb (hunk 7 above)
Comment (ogarocious) [reply db/migrate/20220103143414_add_videoids_to_artists.rb]: working on this!

File: app/views/dance_styles/show.html.erb (hunk 5 above)
Comment (ogarocious) [reply app/views/dance_styles/show.html.erb]: fixed!

File: app/views/dance_styles/show.html.erb (hunk 4 above)
Comment (ogarocious) [reply app/views/dance_styles/show.html.erb]: fixed!

File: app/views/artists/show.html.erb (hunk 3 above)
Comment (ogarocious) [reply app/views/artists/show.html.erb]: fixed!

File: app/views/artists/show.html.erb (hunk 2 above)
Comment (ogarocious) [reply app/views/artists/show.html.erb]: I removed the dash!

File: app/controllers/artists_controller.rb [hunk 8]
```diff
@@ -1,16 +1,33 @@
...
+      format.js
+    end
   end
 
   def new
     @artist = Artist.new
   end
 
   def show
+    puts @artist
```
Comment (melchua) [inline app/controllers/artists_controller.rb:30]: This probably can be removed now right? Looks like it was for debugging purposes.

File: app/controllers/artists_controller.rb (hunk 1 above)
Comment (melchua) [reply app/controllers/artists_controller.rb:65]: Hm. looks like you still have these 3 different video ids as required? I thought you had added a videos array?

**Review by melchua — APPROVED** [review]

**Pull request merged**